GeosCore/tomas_mod.F90

```diff
@@ -6096,68 +6211,328 @@ SUBROUTINE AERODIAG( PTYPE, I, J, L, Nk, Nkd, Mk, Mkd, BOXVOL, State_Grid )
-                !tempsum = tempsum + MK(K,JS) - MKD(K,JS)
-                !IF (PTYPE == 2 ) ACCUM(JS) = ACCUM(JS) + MK(K,JS) - MKD(K,JS)
-             ENDDO
-          ENDDO
-          !IF ( IT_IS_NAN(ACCUM(1))) print *,'ADIAG: Nan',I,J,L
+       IF ( PTYPE == 18 ) THEN
+       DO K=1,IBINS
+           State_Diag%TomasMNFIXcheck2mass(I,J,L,K) = 0.e+0_fp  
+           State_Diag%TomasMNFIXcheck2number(I,J,L,K) = 0.e+0_fp  
+       DO JS = 1, ICOMP-IDIAG
+       IF ( State_Diag%Archive_TomasMNFIXcheck2mass ) THEN
+             S = State_Diag%Map_TomasMNFIXcheck2mass%id2slot(K)
+             IF ( S > 0 ) THEN
+                State_Diag%TomasMNFIXcheck2mass(I,J,L,K) = & 
+                  State_Diag%TomasMNFIXcheck2mass(I,J,L,K) + (MK(K,JS) - MKD(K,JS))/ &
+                         DTCHEM / BOXMASS  ! kg/kg air/sec                      
+             ENDIF
+          ENDIF
+       ENDDO
+       IF ( State_Diag%Archive_TomasMNFIXcheck2number ) THEN
+             S = State_Diag%Map_TomasMNFIXcheck2number%id2slot(K)
+             IF ( S > 0 ) THEN
+                State_Diag%TomasMNFIXcheck2number(I,J,L,K) = & 
+                  State_Diag%TomasMNFIXcheck2number(I,J,L,K) + (NK(K) - NKD(K))/ &
+                         DTCHEM / BOXMASS  ! no./kg air/sec                      
+             ENDIF
+          ENDIF
+       ENDDO
+       ENDIF
 
-          ! Put the updated values back into the diagnostic arrays
-          SELECT CASE ( PTYPE )
-          CASE ( 1 )                ! Condensation diagnostic
-             AD60_COND(1,J,L,:) = ADXX(:)
+       IF ( PTYPE == 19 ) THEN
+       DO K=1,IBINS
+           State_Diag%TomasMNFIXcheck3mass(I,J,L,K) = 0.e+0_fp  
+           State_Diag%TomasMNFIXcheck3number(I,J,L,K) = 0.e+0_fp  
+       DO JS = 1, ICOMP-IDIAG
+       IF ( State_Diag%Archive_TomasMNFIXcheck3mass ) THEN
+             S = State_Diag%Map_TomasMNFIXcheck3mass%id2slot(K)
+             IF ( S > 0 ) THEN
+                State_Diag%TomasMNFIXcheck3mass(I,J,L,K) = & 
+                  State_Diag%TomasMNFIXcheck3mass(I,J,L,K) +( MK(K,JS) - MKD(K,JS))/ &
+                         DTCHEM / BOXMASS  ! kg/kg air/sec                      
+             ENDIF
+          ENDIF
+       ENDDO
+       IF ( State_Diag%Archive_TomasMNFIXcheck3number ) THEN
+             S = State_Diag%Map_TomasMNFIXcheck3number%id2slot(K)
+             IF ( S > 0 ) THEN
+                State_Diag%TomasMNFIXcheck3number(I,J,L,K) = & 
+                  State_Diag%TomasMNFIXcheck3number(I,J,L,K) + (NK(K) - NKD(K))/ &
+                         DTCHEM / BOXMASS  ! no/kg air/sec                      
+             ENDIF
+          ENDIF
+       ENDDO
+       ENDIF
 
-          CASE ( 2 )                ! Coagulation diagnostic
-             AD60_COAG(1,J,L,:) = ADXX(:)
 
-          CASE ( 3 )                ! Nucleation diagnostic
-             AD60_NUCL(1,J,L,:) = ADXX(:)
+    ! PTYPE = 7 is for ND61  --- NOW use for Nucleation at species NK1
```

Comment:
Obsolete BPCH diagnostic code should be deleted instead of commented out to reduce clutter and avoid confusion.

Reply:
Now resolved in commit [c4a3708](https://github.com/geoschem/geos-chem/pull/1378/commits/c4a37080b06dbe9c228fffce228e231ae010ccbc).